pkg/query-service/app/querier/v2/helper.go

```diff
@@ -138,13 +210,36 @@ func (q *querier) runBuilderQuery(
 				}
 				return
 			}
+
+			filteredSeries, startTime, endTime := filterSeriesPoints(series, miss.Start, miss.End, builderQuery.StepInterval)
+
+			// making sure that empty range doesn't doesn't enter the cache
+			if len(filteredSeries) > 0 {
+				filteredMissedSeries = append(filteredMissedSeries, querycache.CachedSeriesData{
+					Data:  filteredSeries,
+					Start: startTime,
+					End:   endTime,
+				})
+			}
```

Comment:
Why do we have an additional check here? If the original query returned empty results, the filteredSeries would list with zero entries?

Reply:
filteredSeries will return empty series, but we don't want overall empty series to be appended as well.

so lets say filteredSeries is empty, if we don't add an if condition the append function will still append it.

Reply:
> but we don't want overall empty series to be appended as well.

Why? A query could return an empty result which is it's actual result that should be cached?

Reply:
Aa right, so this is the case when the actual series lenth is 0. I have added an or condition.